pkg/api/handlers/compat/images_build.go

```diff
@@ -578,8 +578,17 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		pullPolicy = buildahDefine.PolicyMap[query.PullPolicy]
 	} else {
 		if _, found := r.URL.Query()["pull"]; found {
-			if query.Pull {
+			if query.Pull == "false" {
+				pullPolicy = buildahDefine.PullNever
+			} else if query.Pull == "true" {
 				pullPolicy = buildahDefine.PullAlways
+			} else {
+				policyFromMap, foundPolicy := buildahDefine.PolicyMap[query.Pull]
+				if foundPolicy {
+					pullPolicy = policyFromMap
+				} else {
+					utils.BadRequest(w, "pull", query.Pull, fmt.Errorf("invalid pull policy: %q", query.Pull))
```

Comment:
missing return statement, and just guessing but I think the linter would love to see a switch case here but CI will tell you that anyway.